qiskit_nature/properties/second_quantization/second_quantized_property.py

```diff
@@ -13,66 +13,54 @@
 """The SecondQuantizedProperty base class."""
 
 from abc import abstractmethod
-from typing import Any, List, TypeVar, Union
+from typing import Any, List, Type, TypeVar, Union
 
 from qiskit_nature import QiskitNatureError
-from qiskit_nature.drivers import QMolecule as LegacyQMolecule
-from qiskit_nature.drivers import WatsonHamiltonian as LegacyWatsonHamiltonian
-from qiskit_nature.drivers.second_quantization import QMolecule, WatsonHamiltonian
+from qiskit_nature.drivers import QMolecule, WatsonHamiltonian
 from qiskit_nature.operators.second_quantization import SecondQuantizedOp
 
 from ..grouped_property import GroupedProperty
 from ..property import Property
 
-LegacyElectronicStructureDriverResult = Union[QMolecule, LegacyQMolecule]
-LegacyVibrationalStructureDriverResult = Union[WatsonHamiltonian, LegacyWatsonHamiltonian]
-LegacyDriverResult = Union[
-    LegacyElectronicStructureDriverResult, LegacyVibrationalStructureDriverResult
-]
+LegacyDriverResult = Union[QMolecule, WatsonHamiltonian]
 
 
 class SecondQuantizedProperty(Property):
     """The SecondQuantizedProperty base class.
 
-    A second-quantization property provides the logic to transform a raw data (as e.g. produced by a
+    A second-quantization property provides the logic to transform raw data (as e.g. produced by a
     `qiskit_nature.second_quantization.drivers.BaseDriver`) into a
     `qiskit_nature.operators.second_quantization.SecondQuantizedOp`.
```

Comment:
```suggestion
    A second-quantization property provides the logic to transform raw data (as e.g. produced by a
    :class:`~qiskit_nature.second_quantization.drivers.BaseDriver`) into a
    :class:`~qiskit_nature.operators.second_quantization.SecondQuantizedOp`.
```
So these turn out as links in the doc rather than just highlighted text - and limited now to just the class name since it does not need to be so verbose with a link I think.

Reply:
I see there are other places in the text that would benefit from being links rather than quoted text as they refer to classes etc that it would be nice to navigate too. Since the text existed from before I cannot annotate it like above as its too far from any changes. And rather than try to include them here I will just note this. It would be nice to have them as links but it can always be improved later if it cannot be done now

Reply:
I have iterated through all files in the `qiskit_nature.properties` submodule looking for "`" characters.
I think the situation should be generally better now :+1: